set-web/src/main/java/eu/europeana/set/web/service/authorization/AuthorizationServiceImpl.java

```diff
@@ -1,124 +1,102 @@
-	}
-
-	//verify client app privileges 
-	protected boolean hasPermission(Application app, String setId, String operationName) {
-		if(Operations.MODERATION_ALL.equals(operationName) 
-				|| Operations.RETRIEVE.equals(operationName) 
-				|| Operations.UPDATE.equals(operationName) 
-				|| Operations.CREATE.equals(operationName))
-			return true;
-		
-		return setId != null; 
-	}
-
-	//verify user privileges
-	protected boolean hasPermission(Agent user, String operationName) {
-		Roles role = Roles.valueOf(user.getUserGroup());
-		
-		for (String operation : role.getOperations()) {
-			if(operation.equalsIgnoreCase(operationName))
-				return true;//users is authorized, everything ok
-		}
-		
-		return false;
-	}
-
-	protected boolean isAdmin(Agent user) {
-		return Roles.ADMIN.name().equals(user.getUserGroup());
-	}
-	
-	protected boolean isTester(Agent user) {
-		return Roles.TESTER.name().equals(user.getUserGroup());
-	}
-	
-	public UserSetConfiguration getConfiguration() {
-		return configuration;
-	}
-
-	public void setConfiguration(UserSetConfiguration configuration) {
-		this.configuration = configuration;
-	}
-}
+package eu.europeana.set.web.service.authorization;
+
+import javax.annotation.Resource;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.security.oauth2.provider.ClientDetailsService;
+
+import eu.europeana.api.commons.definitions.vocabulary.Role;
+import eu.europeana.api.commons.service.authorization.BaseAuthorizationService;
+import eu.europeana.set.definitions.config.UserSetConfiguration;
+import eu.europeana.set.definitions.model.agent.Agent;
+import eu.europeana.set.definitions.model.authentication.Application;
+import eu.europeana.set.web.model.vocabulary.Operations;
+import eu.europeana.set.web.model.vocabulary.Roles;
+import eu.europeana.set.web.model.vocabulary.UserRoles;
+import eu.europeana.set.web.service.authentication.AuthenticationService;
+
+
+public class AuthorizationServiceImpl extends BaseAuthorizationService implements AuthorizationService {
+	
+	protected final Logger logger = LogManager.getLogger(getClass());
+
+	@Resource
+	AuthenticationService authenticationService;
+	
+	@Resource
+	UserSetConfiguration configuration;
+
+    @Resource(name = "commons_oauth2_europeanaClientDetailsService")
+    ClientDetailsService clientDetailsService;
+	
+	public AuthorizationServiceImpl(AuthenticationService authenticationService){
+		this.authenticationService = authenticationService;
+	}
+	
+	public AuthorizationServiceImpl(){
+		
+	}
+	
+	public AuthenticationService getAuthenticationService() {
+		return authenticationService;
+	}
+
+	public void setAuthenticationService(AuthenticationService authenticationService) {
+		this.authenticationService = authenticationService;
+	}
+	
+	//verify user privileges
+	protected boolean hasPermission(Agent user, String operationName) {

```

Comment:
is this method still used, if not, please remove it

Reply:
Please also remove isAdmin adn isTester methods if they are not used anymore. Check also the Agent class if it is used

Reply:
removed mentioned methods. Agent ist used in almost 400 cases.